components/site/src/lib.rs

```diff
@@ -749,7 +749,13 @@ impl Site {
         start = log_time(start, "Rendered sitemap");
 
         let library = self.library.read().unwrap();
-        if self.config.generate_feed {
+        if self.config.generate_feed
+            || self
+                .config
+                .languages
+                .get(&self.config.default_language)
+                .map_or(false, |lang_opt| lang_opt.generate_feed)
```

Comment:
looks like we do need better config querying based on a language like you did with `language_feed_filename`.
No need to change anything here, just thinking aloud

Reply:
Let me know if the change I have is what you were thinking.